python/lsst/sims/utils/CodeUtilities.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+
+def _validate_inputs(input_list, input_names, method_name):
+    """
+    This method will validate the inputs of other methods.
+
+    input_list is a list of the inputs passed to a method.
+
+    input_name is a list of the variable names associated with
+    input_list
+
+    method_name is the name of the method whose input is being validated.
+
+    _validate_inputs will verify that all of the inputs in input_list are:
+
+    1) of the same type
+    2) either floats or numpy arrays
+    3) if they are numpy arrays, they all have the same length
+
+    If any of these criteria are violated, a RuntimeError will be raised
+
+    returns True if the inputs are numpy arrays; False if not
+    """
+
+    if isinstance(input_list[0], np.ndarray):
+        desired_type = np.ndarray
+    elif isinstance(input_list[0], np.float):
+        desired_type = np.float
```

Comment:
Instead of comparing to float (could the number be an integer and still be valid?) would it be better to compare against numbers.Number? (isinstance(input_list[0], Number) 

>>> from numbers import Number
>>> a = 5.2
>>> isinstance(a, Number)
True
>>> a = 5
>>> isinstance(a, Number)
True
>>> a = '5'
>>> isinstance(a, Number)
False